Change Metadata schema value from `Dict` to `Any`

donatoaz opened the pull request:
According to [AWS docs on the format of Metadata for a CloudFormation Template](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/metadata-section-structure.html) there is no hard requirement that values in the `Metadata` object be objects themselves. They can take any value.

This PR proposes to change `Metadata` schema to accept `Any` instead of a `Dict` such that the following metadata values are all acceptable:

```
"Metadata": {
    "Foo": "a_string_value",
    "Bar": 1
    "Fizz": ["an", "array"],
    "Buzz": { "an": "object" }
  },
```

ocrawford555 reviewed (approved):
LGTM, thanks for the contribution!

jsoucheiron reviewed (approved):
LGTM, thanks for contributing!
We'll add this in the next release

donatoaz commented:
Hi! I've updated the `pycfmodel/cloudformation_actions.py` so tests would pass, and added two new test cases to `tests/test_cf_model.py`.

Let me know if you agree
